### ISSUE TITLE
Transform method

### ISSUE THREAD

**Issue by crhea93** [sashakolpakov/dire-jax#19]
The fit_transform() method works very nicely on some custom training data. However, there does not seem to be a way to then apply this to test data. I note that there is a transform() method, but it does not take any arguments. Would it be possible to use it like the transform() method in other dimensionality reduction algorithms in sklearn (for example [https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html#sklearn.decomposition.PCA.transform](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html#sklearn.decomposition.PCA.transform)). 

Just for an example on how nicely this is working on my test data

![Image](https://github.com/user-attachments/assets/5ea4b5ed-e7e2-4bf4-967e-f605eea475bf)

**Comment by crhea93** [comment]
So I'm working with contrastive learning embeddings of astronomy spectra with a continuous label that goes from -2 to 2. The nice piece about the DiRe embeddings is that they seem to carry more information than the tSNE embeddings. It would be possible to test this by running a zero shot algorithm like KNN on both and computing a metric and comparing those values. Visually, it looks like the DiRe embeddings would be more informative.

**Comment by crhea93** [comment]
I also see that the tSNE implementation in scikit does not seem to have a transform method either! That being said, I think it would be an extremely useful addition even if it did some type of interpolation.

**Comment by sashakolpakov** [comment]
> That being said, I think it would be an extremely useful addition ...

@crhea93 Is this a mostly general statement, or it means that having transform() will make DiRe immediately usable for your research, or any other concrete project? Having already spent considerable time on DiRe, I would be happy to know it gets applied, not just published and largely forgotten.

**Comment by sashakolpakov** [comment]
@crhea93 @lmcinnes Thank you for your time and comments!
